frame/nfts/src/lib.rs

```diff
@@ -1768,6 +1789,35 @@ pub mod pallet {
 				witness_price,
 			)
 		}
+
+		/// Mint an item by providing the pre-signed approval.
+		///
+		/// Origin must be Signed.
+		///
+		/// - `data`: The pre-signed approval that consists of the information about the item, its
+		///   metadata, attributes, who can mint it (`None` for anyone) and until what block number.
+		/// - `signature`: The signature of the `data` object.
+		/// - `signer`: The `data` object's signer. Should be an owner of the collection.
+		///
+		/// Emits `Issued` on success.
+		/// Emits `ItemMetadataSet` if the metadata was not empty.
+		#[pallet::call_index(37)]
+		#[pallet::weight(T::WeightInfo::mint_pre_signed(data.attributes.len() as u32))]
+		pub fn mint_pre_signed(
+			origin: OriginFor<T>,
+			data: PreSignedMintOf<T, I>,
+			signature: MultiSignature,
+			signer: MultiSigner,
+		) -> DispatchResult {
+			let origin = ensure_signed(origin)?;
+			let msg = Encode::encode(&data);
```

Comment:
I think it should be generally Okay to sign the encoded `PreSignedMint`.  
Sr25519 Substrate signatures have `substrate` as Merlin context, not sure about the other signature schemes. We could still prefix the payload with `nfts`, just to be sure.  
@burdges